packages/wdio-config/src/node/ConfigParser.ts

```diff
@@ -259,11 +269,11 @@ export default class ConfigParser {
         // when CLI --spec is explicitly specified, this._config.specs contains the filtered
         // specs matching the passed pattern else the specs defined inside the config are returned
         let specs = ConfigParser.getFilePaths(this._config.specs!, this._config.rootDir, this._pathService)
-        let exclude = ConfigParser.getFilePaths(this._config.exclude!, this._config.rootDir, this._pathService)
+        let exclude = allKeywordsContainPath(this._config.exclude!) ? ConfigParser.getFilePaths(this._config.exclude!, this._config.rootDir, this._pathService) : this._config.exclude!
         const suites = Array.isArray(this._config.suite) ? this._config.suite : []
 
         // only use capability excludes if (CLI) --exclude or config exclude are not defined
-        if (Array.isArray(capExclude) && exclude.length === 0){
+        if (Array.isArray(capExclude) && exclude!.length === 0){
```

Comment:
Added a `|| []` above
```suggestion
        if (Array.isArray(capExclude) && exclude.length === 0){
```

Reply:
Done